torch/csrc/jit/script/compiler.cpp

```diff
@@ -694,7 +699,11 @@ struct to_ir {
     }
     for(;it != end; ++it) {
       auto& name = (*it).ident().name();
-      arguments.push_back({name, DynamicType::get()});
+      TypePtr arg_type = DynamicType::get();
+      if (def_and_types.arg_types.size()) {
```

Comment:
If the FunctionSchema is optional, it should be represented as `at::optional<FunctionSchema>`